### PR TITLE
Implement Responsive header

### LINKED ISSUE #210: Fix responsiveness of Header for Mobile
**Describe the bug**
The current Header is not responsive yet.

**Expected behavior**
To include a hamburger menu for mobile phone displaying all the navitems

**Screenshots**
image

**Smartphone (please complete the following information):**
 - Device: Phone
 - OS: Android
 - Browser Chrome
 - Version latest

### PR DESCRIPTION
**Pull request by harsh3dev**
Resolves #210 

- [x] Implement clean and minimal code by replacing reused code block.
- [x] Added hamburger menu for mobile devices which is expandable for more items.
- [x] Highlight the current active page on the header.
- [x] Fix the overall style and responsiveness of the component.

## Mobile View

https://github.com/user-attachments/assets/63c4f744-2aa4-44a6-a743-3fa260e4ead8

---
## Desktop view
image

### CODE REVIEW

**Review by kasya — CHANGES_REQUESTED** [review]
This doesn't look like your videos for me at all 🤔 . This is projects page on iPhone 14 pro max. The 
Screenshot 2024-12-21 at 3 56 12 PM

The theme toggle get's moved to the next line in the header. 
Also, the meta items in the card component get messed up.

I'm thinking because of the style of this card we should probably show these data items differently on the mobile devices. We can try showing them as a list under the Card title/project title. Right now this doesn't work (this is contribute page on iPhone 12):
Screenshot 2024-12-21 at 4 02 50 PM

**Comment by harsh3dev** [comment]
@kasya Thanks for the suggestions on responsiveness of the Card
For the header, are you having a look at the old FE? As in the image which you mentioned it seems it is on `localhost:8000` Please review the new header at `localhost:3000` which is for the new FE.
Thank you

**Comment by kasya** [comment]
> @kasya Thanks for the suggestions on responsiveness of the Card For the header, are you having a look at the old FE? As in the image which you mentioned it seems it is on `localhost:8000` Please review the new header at `localhost:3000` which is for the new FE. Thank you

@harsh3dev  Oh man.. Sorry about that! I'll review it again asap!

**Review by kasya — CHANGES_REQUESTED** [review]
@harsh3dev could you please merge *main* in and resolve conflicts? 🙏🏼

**Comment by harsh3dev** [comment]
@kasya Sorry for the late reply it took me some time to get back to you.
I have resolved all the conflicts.

**Review by kasya — CHANGES_REQUESTED** [review]
Hello @harsh3dev ! Thanks for working on this! I reviewed the latest updates and made a video of the issue that I noticed. Depending on the page we are on  - the hamburger icon is not always visible and accessible.

https://github.com/user-attachments/assets/e96a6966-06cf-4464-97b6-6aa6964f248d


We also have another [PR](https://github.com/OWASP/Nest/pull/252) working on the same thing (not sure how that happened that both of you were working on the same thing at the same time). I also left some comments there to address. I do like an extra thing that was implemented there for footer responsivenes, so would love to merge in at least some of those changes too.

As for your PR if you can update this issue with the Header quickly - we can use your header updates and use the footer and card updates from that another PR. 👌🏼

**Comment by harsh3dev** [comment]
Uploading Recording 2024-12-26 200249.mp4…

I have fixed the shifting of the header.
I guess the header in [PR](https://github.com/OWASP/Nest/pull/252) is redundant work as I have implemented the new header before it.
When the card as well as the footer will be merged from #252, the UI will look amazing.
Thanks for your valuable review.

File: frontend/src/components/Header.tsx [hunk 1]
```diff
@@ -1,79 +1,113 @@
-import { faMoon, faSun } from '@fortawesome/free-regular-svg-icons'
```
Comment (kasya) [inline frontend/src/components/Header.tsx:1]: Why have we changed this? 🤔 We kinda wanted to have these sun and moon icons for dark and light theme. Can we go back to a previous view that used these icons? The `ModeToggle` component is not something we use right now, but we'd like to keep to update later on.

File: frontend/src/components/Header.tsx [hunk 2]
```diff
@@ -1,79 +1,113 @@
...
-                <i className="fa-regular fa-moon"></i>
-              </span>
-            </label>
-          </div>
+          {headerLinks.map((link, i) => (
+            <NavLink
+              key={i}
+              to={link.href}
+              className={cn(
+                'block px-3 py-2 text-slate-700 hover:text-slate-800 dark:text-slate-300 dark:hover:text-slate-200',
```
Comment (kasya) [inline frontend/src/components/Header.tsx]: Also, can we put the `navlink` class back? it is a custom class that addresses link color for navigation. Right now active link looks like this when it's active in light theme. This is a really weird looking color. Can we just make it heavier/darker grey color? Similar how it is in a dark theme with bolder text for active link?
Screenshot 2024-12-26 at 4 05 10 PM

**Review by kasya — CHANGES_REQUESTED** [review]
@harsh3dev Thanks for addressing this! This works great now! A couple of comments about the UI updates.

File: frontend/src/components/Header.tsx (hunk 2 above)
Comment (harsh3dev) [reply frontend/src/components/Header.tsx]: I have added the navlink now

File: frontend/src/components/Header.tsx (hunk 1 above)
Comment (harsh3dev) [reply frontend/src/components/Header.tsx:1]: I have re-added the sun and moon icon in the header

File: frontend/src/components/ModeToggle.tsx [hunk 3]
```diff
@@ -26,14 +27,14 @@ function ModeToggle({ className }: { className?: string }) {
   return (
     <div className={cn('flex items-center space-x-2', className)}>
       <span className="text-sm font-medium text-gray-900 dark:text-gray-100">
-        <FontAwesomeIconWrapper icon="fa-regular fa-lightbulb" />
+        <FontAwesomeIcon icon={faSun} />
```
Comment (kasya) [inline frontend/src/components/ModeToggle.tsx]: Could you update this to have only one icon and show relevant based on theme? Same we have it right now it [staging](https://nest.owasp.dev/projects/contribute/)?
```suggestion
        <FontAwesomeIcon icon={dark ? faMoon : faSun} />
```

File: frontend/src/components/ModeToggle.tsx [hunk 4]
```diff
@@ -26,14 +27,14 @@ function ModeToggle({ className }: { className?: string }) {
...
       </span>
       <label className="relative inline-flex cursor-pointer items-center">
         <input type="checkbox" className="peer sr-only" checked={dark} onChange={darkModeHandler} />
         <div className="peer h-6 w-11 rounded-full bg-gray-200 after:absolute after:left-[2px] after:top-[2px] after:h-5 after:w-5 after:rounded-full after:border after:border-gray-300 after:bg-white after:transition-all after:content-[''] peer-checked:after:translate-x-full peer-checked:after:border-white dark:border-gray-600 dark:bg-gray-700"></div>
       </label>
       <span className="text-sm font-medium text-gray-900 dark:text-gray-100">
```
Comment (kasya) [inline frontend/src/components/ModeToggle.tsx]: We can remove this whole `span` element too.

**Review by kasya — CHANGES_REQUESTED** [review]
Almost there! Left a suggestion and a comment

File: frontend/src/components/ModeToggle.tsx (hunk 3 above)
Comment (harsh3dev) [reply frontend/src/components/ModeToggle.tsx]: Okay I understood,  we need either moon or the sun icon, sure ill implement this.

**Comment by harsh3dev** [comment]
@kasya I have removed the extra span and made only one icon to show at a time and made It look similar to the current one.

**Review by kasya — APPROVED** [review]
Nice work! Thank you 👍🏼

**Pull request merged**